Add report interaction calls

andresmgot opened the pull request:
Add report interaction for the following events:

 - Toggle a group (e.g. "high severity" section or a specific check). 
 - Interaction with a check. Report the interaction type (e.g. `refresh_clicked`), the check type (e.g. `datasource`) and the step ID (e.g. `health_check`).
 - Global actions (e.g. refresh or delete the report).
 
 Fixes https://github.com/grafana/grafana-community-team/issues/484

andresmgot committed:
Add report interaction calls

andresmgot committed:
less comments

andresmgot committed:
fix test

sympatheticmoose reviewed (commented):
Some feedback from manual testing:

**%group_toggled**
- seems to include a lot of random properties we don't need i.e. grafana version, plugin_id, plugin_name, plugin_type, plugin_version.
- would be helpful to know if it is an open or close action on the toggle
- no event on expanding "More info"?

**%check_interaction**
-  silence_clicked means we can't tell if the check is being silenced or un-silenced
 - Clicking documentation - the CTA for license check fires the event as toggling the group (and does so) - feels like a bug or we should have "View documentation" as the fix button
<img width="1257" height="317" alt="Screenshot 2025-07-18 at 16 36 51" src="https://github.com/user-attachments/assets/df1c7f2e-94a9-4570-9824-2f22bd08e5d3" /> 

Completely unrelated feedback - I wonder if we should shift configure out of More Info, maybe next to refresh/purge, as I do agree with the event name that it is a global action - but doesn't really appear as one.

andresmgot commented:
> Some feedback from manual testing:
> 
> **%group_toggled**
> 
> * seems to include a lot of random properties we don't need i.e. grafana version, plugin_id, plugin_name, plugin_type, plugin_version.

That's provided by the exposed hook `usePluginInteractionReporter`, it cannot be avoided.

> * would be helpful to know if it is an open or close action on the toggle

👍  added

> * no event on expanding "More info"?

👍  added

> 
> **%check_interaction**
> 
> * silence_clicked means we can't tell if the check is being silenced or un-silenced

Added a property to say if it's silenced or not.

> * Clicking documentation - the CTA for license check fires the event as toggling the group (and does so) - feels like a bug or we should have "View documentation" as the fix button

Interesting bug, fixed 👍 

> Completely unrelated feedback - I wonder if we should shift configure out of More Info, maybe next to refresh/purge, as I do agree with the event name that it is a global action - but doesn't really appear as one.

I can do this in another PR, it may make more sense that what we have now 👍

andresmgot committed:
apply feedback

andresmgot committed:
tests

academo commented on src/components/Actions/Actions.tsx:
```diff
@@ -16,9 +17,22 @@ export default function Actions({ isCompleted, checkStatuses }: ActionsProps) {
   const styles = useStyles2(getStyles);
 
+  const handleRefreshClick = (e: React.MouseEvent) => {
+    e.preventDefault();
```
is this preventDefault really necessary for this interaction?

academo reviewed (commented):
Generally speaking:

* Use ENUMS instead of union types for the events
* Consider if you really need separate tracking methods because they are merely helpers to construct a map, maybe you can make due with a single tracking method and pass custom attributes.

tolzhabayev reviewed (approved):
lgtm for the tracking naming convention

andresmgot committed:
apply feedback

andresmgot commented:
Thanks for the review!

> * Consider if you really need separate tracking methods because they are merely helpers to construct a map, maybe you can make due with a single tracking method and pass custom attributes.

Slight preference here to have separate tracking methods, feels a bit more organized in buckets.
